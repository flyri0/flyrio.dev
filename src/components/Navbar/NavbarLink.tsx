export interface NavbarLinkProps {
  content: string;
  href: string;
}

export default function NavbarLink({
  content,
  href,
}: NavbarLinkProps): JSX.Element {
  return (
    <li className="transition hover:text-shadow hover:underline">
      <a href={href}>{content}</a>
    </li>
  );
}
