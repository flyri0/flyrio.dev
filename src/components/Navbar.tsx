import NavbarLink from '@components/NavbarLink';

export default function Navbar() {
  return (
    <nav className="default-margin absolute flex w-full items-center justify-between">
      <a href="/">
        <img
          src="/logo.svg"
          alt="Um logotipo que é letra F composta por figuras hexágonais."
          className="w-10"
        />
      </a>

      <div className="flex flex-col md:flex-row md:gap-x-5">
        <NavbarLink targetSection="about">Sobre Mim</NavbarLink>
        <NavbarLink targetSection="projects">Projetos</NavbarLink>
        <NavbarLink targetSection="contact">Contato</NavbarLink>
      </div>
    </nav>
  );
}
