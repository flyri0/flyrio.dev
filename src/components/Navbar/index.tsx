import NavbarLink from '@components/Navbar/NavbarLink';
import useWindowScrollPosition from '@hooks/useWindowScrollPosition';
import { useEffect, useState } from 'react';

export default function Navbar(): JSX.Element {
  const [isOnTop, setIsOnTop] = useState<boolean | null>(true);
  const [{ y }] = useWindowScrollPosition();

  useEffect(() => {
    if (Number(y) <= 100) {
      setIsOnTop(true);
    } else {
      setIsOnTop(false);
    }
  }, [y, isOnTop]);

  return (
    <nav
      className={`default-margin fixed z-10 flex w-full flex-row items-center justify-between border-b transition duration-300 ${
        isOnTop
          ? 'border-b-transparent bg-transparent'
          : 'border-b-accent-2 bg-background bg-opacity-90 backdrop-blur'
      }`}>
      <a href="/#top">
        <img
          className="w-10 transition hover:-translate-y-0.5"
          src="/brand.svg"
          alt="Letra F composta por figuras hexÃ¡gonais."
        />
      </a>

      <ul className="flex flex-col text-right font-spaceGrotesk font-bold md:flex-row md:gap-5">
        <NavbarLink href="#about" content="Sobre Mim" />
        <NavbarLink href="#projects" content="Projetos" />
        <NavbarLink href="#contact" content="Contato" />
      </ul>
    </nav>
  );
}
