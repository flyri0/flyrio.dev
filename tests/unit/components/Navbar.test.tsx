import { render } from '@testing-library/react';
import Navbar from '@components/Navbar';

describe('<Navbar />', () => {
  it('should match the snapshot', () => {
    const component = render(<Navbar />);

    expect(component).toMatchSnapshot();
  });
});
