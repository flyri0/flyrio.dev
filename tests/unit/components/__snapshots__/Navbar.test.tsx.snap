// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Navbar /> > should match the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="default-margin absolute flex w-full items-center justify-between"
      >
        <a
          href="/"
        >
          <img
            alt="Um logotipo que é letra F composta por figuras hexágonais."
            class="w-10"
            src="/logo.svg"
          />
        </a>
        <div
          class="flex flex-col md:flex-row md:gap-x-5"
        >
          <button
            class="text-right font-spaceGrotesk text-sm font-bold transition hover:text-shadow md:text-center md:text-base"
          >
            Sobre Mim
          </button>
          <button
            class="text-right font-spaceGrotesk text-sm font-bold transition hover:text-shadow md:text-center md:text-base"
          >
            Projetos
          </button>
          <button
            class="text-right font-spaceGrotesk text-sm font-bold transition hover:text-shadow md:text-center md:text-base"
          >
            Contato
          </button>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="default-margin absolute flex w-full items-center justify-between"
    >
      <a
        href="/"
      >
        <img
          alt="Um logotipo que é letra F composta por figuras hexágonais."
          class="w-10"
          src="/logo.svg"
        />
      </a>
      <div
        class="flex flex-col md:flex-row md:gap-x-5"
      >
        <button
          class="text-right font-spaceGrotesk text-sm font-bold transition hover:text-shadow md:text-center md:text-base"
        >
          Sobre Mim
        </button>
        <button
          class="text-right font-spaceGrotesk text-sm font-bold transition hover:text-shadow md:text-center md:text-base"
        >
          Projetos
        </button>
        <button
          class="text-right font-spaceGrotesk text-sm font-bold transition hover:text-shadow md:text-center md:text-base"
        >
          Contato
        </button>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
